packages/canvas/render/src/application-function/global-state.ts

```
@@ -0,0 +1,31 @@
+  const globalState = ref([])
+
+  const setGlobalState = (data = []) => {
+    globalState.value = data
+  }
+  const stores = shallowReactive({})
+  watchEffect(() => {
+    reset(stores)
+    globalState.value.forEach(({ id, state = {}, getters = {} }) => {
+      const computedGetters = Object.keys(getters).reduce(
+        (acc, key) => ({
+          ...acc,
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)